android/src/com/frostwire/android/gui/activities/MainActivity.java

```diff
@@ -204,6 +285,48 @@ public void run() {
         }.start();
     }
 
+    private boolean isShutdown() {
+        return isShutdown(null);
+    }
+
+    private boolean isShutdown(Intent intent) {
+        if (intent == null) {
+            intent = getIntent();
+        }
+        boolean result = intent != null && intent.getBooleanExtra("shutdown-" + ConfigurationManager.instance().getUUIDString(), false);
+        if (result) {
+            shutdown();
+        }
+        return result;
+    }
+
+
+    public MainActivity() {
+        super(R.layout.activity_main);
+//        this.controller = new MainController(this);
+        this.fragmentsStack = new Stack<>();
+        this.permissionsCheckers = initPermissionsCheckers();
+    }
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_SEARCH) {
+            if (!(getCurrentFragment() instanceof SearchFragment)) {
+                controller.switchFragment(R.id.menu_main_search);
+            }
+        } else if (keyCode == KeyEvent.KEYCODE_MENU) {
+//            toggleDrawer();
```

Comment:
hmm, gotta have something like this again.
Although I haven't seen any new androids with this key, users that do have it will feel the pain of not being able to pull out the menu with the menu key.